sydent/replication/pusher.py

```diff
@@ -95,59 +95,54 @@ def doLocalPush(self):
-                push_data = {}
+            # Dictionary for holding all data to push
+            push_data = {}
 
-                # Dictionary for holding all the ids of db tables we've successfully replicated up to
-                ids = {}
-                total_updates = 0
+            # Dictionary for holding all the ids of db tables we've successfully replicated up to
+            ids = {}
+            total_updates = 0
 
-                # Push associations
-                (push_data["sg_assocs"], ids["sg_assocs"]) = self.getSignedAssociationsAfterId(p.lastSentAssocsId, ASSOCIATIONS_PUSH_LIMIT, p.shadow)
-                total_updates += len(push_data["sg_assocs"])
+            # Push associations
+            (push_data["sg_assocs"], ids["sg_assocs"]) = self.getSignedAssociationsAfterId(p.lastSentAssocsId, ASSOCIATIONS_PUSH_LIMIT, p.shadow)
+            total_updates += len(push_data["sg_assocs"])
 
-                # Push invite tokens and ephemeral public keys
-                (push_data["invite_tokens"], ids["invite_tokens"]) = join_token_store.getInviteTokensAfterId(p.lastSentInviteTokensId, INVITE_TOKENS_PUSH_LIMIT)
-                (push_data["ephemeral_public_keys"], ids["ephemeral_public_keys"]) = join_token_store.getEphemeralPublicKeysAfterId(p.lastSentEphemeralKeysId, EPHEMERAL_PUBLIC_KEYS_PUSH_LIMIT)
-                total_updates += len(push_data["invite_tokens"]) + len(push_data["ephemeral_public_keys"])
+            # Push invite tokens and ephemeral public keys
+            (push_data["invite_tokens"], ids["invite_tokens"]) = join_token_store.getInviteTokensAfterId(p.lastSentInviteTokensId, INVITE_TOKENS_PUSH_LIMIT)
+            (push_data["ephemeral_public_keys"], ids["ephemeral_public_keys"]) = join_token_store.getEphemeralPublicKeysAfterId(p.lastSentEphemeralKeysId, EPHEMERAL_PUBLIC_KEYS_PUSH_LIMIT)
+            total_updates += len(push_data["invite_tokens"]) + len(push_data["ephemeral_public_keys"])
 
-                logger.debug("%d updates to push to %s", total_updates, p.servername)
-                if total_updates:
-                    logger.info("Pushing %d updates to %s:%d", total_updates, p.servername, p.port)
+            logger.debug("%d updates to push to %s", total_updates, p.servername)
+            if total_updates:
+                logger.info("Pushing %d updates to %s:%d", total_updates, p.servername, p.port)
+                try:
+                    p.is_being_pushed_to = True
                     updateDeferred = p.pushUpdates(push_data)
                     updateDeferred.addCallback(self._pushSucceeded, peer=p, ids=ids)
                     updateDeferred.addErrback(self._pushFailed, peer=p)
-                    break
-        finally:
-            if not updateDeferred:
-                self.pushing = False
+                except Exception as e:
+                    logger.exception("Error pushing updates to %s", p.servername)
+                    p.is_being_pushed_to = False
```

Comment:
This probably won't actually catch any exceptions that are thrown by `pushUpdates`, since I'd expect them to get thrown into the `_pushFailed` errback.